nixos/modules/services/databases/mysql.nix

```diff
@@ -319,6 +321,7 @@ in
                 mkdir -m 0700 -p ${cfg.dataDir}
                 chown -R ${cfg.user} ${cfg.dataDir}
                 ${mysql}/bin/mysql_install_db --defaults-file=/etc/my.cnf ${installOptions}
+                ${optionalString (cfg.replication.role == "master" || cfg.replication.role == "slave") "rm ${cfg.dataDir}/mysql-bin-${toString cfg.replication.serverId}.*"}
```

Comment:
I'm not so sure about whether we should it like that.

Of course, it'll only get triggered when there was no "mysql" database before, maybe to fixup something done "wrongly" in the `mysql_install_db` script.
https://github.com/NixOS/nixpkgs/pull/51314#issuecomment-493660001 suggests that `mysql_install_db` should be run without having `log-bin=mysql-bin` specified, and this command seems to remove the binary log - but I'm no mysql expert here.

One cleaner way to do this would be to to run `mysql_install_db` with a custom configuration that doesn't enable the binary log (but all other options).

Is there some upstream documentation that at least briefly mentions the issues we're experiencing? Or is the behaviour a bug upstream, and running mysql without `log-bin` for `mysql_install_db` a workaround?

Maybe we only need to disable the binary log when installing the slave nodes?

I'd be nice if our module would support changing from a single node mysql server to one with some slave nodes, so it might be worth investing some time here…

Reply:
also cc @grahamc here.

Reply:
I think that after creating the database using `mysql_install_db --defaults-file=/etc/ my.cnf`, the files `mysql-bin-*.00000*` do not contain critical data. Deleting these files should not affect server working.
When you first start mysql, the file contents `mysql-bin-*.00000*` identical. And when replicating gives an error.